platformio.ini

```diff
@@ -9,79 +9,159 @@
 lib_extra_dirs = 
     src/_micro-api/libraries/
 
-[env:nanoCC1101@debug]
+; general - for all devices, using ${env.***}
+
+; Unix, example
+monitor_port = /dev/ttyUSB0
+upload_port = /dev/ttyUSB0
+
+; Windows, example
+; monitor_port = COM4
+; upload_port = COM4
+
+[env:nano_CC1101_bootloader_old@debug]
+; bootloader ATmegaBOOT
 platform = atmelavr
 board = nanoatmega328
 monitor_speed = 57600
 framework = arduino
-monitor_port = COM9
+monitor_port = ${env.monitor_port}
+upload_port = ${env.upload_port}
 build_flags=-D OTHER_BOARD_WITH_CC1101=1 -D DEBUG
 
-[env:nanoCC1101]
+[env:nano_CC1101_bootloader_old]
+; bootloader ATmegaBOOT
 platform = atmelavr
 board = nanoatmega328
 monitor_speed = 57600
 framework = arduino
-monitor_port = COM4
+monitor_port = ${env.monitor_port}
+upload_port = ${env.upload_port}
 build_flags=-D OTHER_BOARD_WITH_CC1101=1
 
-[env:nano328@debug]
+[env:nano_CC1101_bootloader_new@debug]
+; bootloader Optiboot
+platform = atmelavr
+board = nanoatmega328new
+monitor_speed = 57600
+framework = arduino
+monitor_port = ${env.monitor_port}
+upload_port = ${env.upload_port}
+build_flags=-D OTHER_BOARD_WITH_CC1101=1 -D DEBUG
+
+[env:nano_CC1101_bootloader_new]
+; bootloader Optiboot
+platform = atmelavr
+board = nanoatmega328new
+monitor_speed = 57600
+framework = arduino
+monitor_port = ${env.monitor_port}
+upload_port = ${env.upload_port}
+build_flags=-D OTHER_BOARD_WITH_CC1101=1
+
+[env:nano328_bootload_old@debug]
+; bootloader ATmegaBOOT
 platform = atmelavr
 board = nanoatmega328
 monitor_speed = 57600
 framework = arduino
-monitor_port = COM9
+monitor_port = ${env.monitor_port}
+upload_port = ${env.upload_port}
 build_flags=-D DEBUG
 src_filter =-cc1101.h -cc1101.cpp +<*> -<.git/> -<.svn/> -<example/> -<examples/> -<arduino-ide/SIGNALDuino/> -<_micro-api/libraries/ArduinoJson/test/> -<_micro-api/libraries/ArduinoJson/third-party/> -<_micro-api/libraries/fastdelegate/examples/>-<test/> -<tests/> -<_micro-api/libraries/ArduinoJson/fuzzing/>
 
-[env:nano328]
+[env:nano328_bootload_old]
+; bootloader ATmegaBOOT
 platform = atmelavr
 board = nanoatmega328
 monitor_speed = 57600
 framework = arduino
-monitor_port = COM9
+monitor_port = ${env.monitor_port}
+upload_port = ${env.upload_port}
 build_flags=
 
+[env:nano328_bootload_new@debug]
+; bootloader Optiboot
+platform = atmelavr
+board = nanoatmega328new
+monitor_speed = 57600
+framework = arduino
+monitor_port = ${env.monitor_port}
+upload_port = ${env.upload_port}
+build_flags=-D DEBUG
+src_filter =-cc1101.h -cc1101.cpp +<*> -<.git/> -<.svn/> -<example/> -<examples/> -<arduino-ide/SIGNALDuino/> -<_micro-api/libraries/ArduinoJson/test/> -<_micro-api/libraries/ArduinoJson/third-party/> -<_micro-api/libraries/fastdelegate/examples/>-<test/> -<tests/> -<_micro-api/libraries/ArduinoJson/fuzzing/>
```

Comment:
Meinst Du das passt hier oder brauchen wir das für alle?

Reply:
Was meinst du mit beötigen wir für alle?

Solle es um 
- den Bootloader gehen, dieser kann je User und Alter der HW variieren.

- die Variable monitor_port / upload_port gehen, so ist es sehr praktisch nur einmal diese zu ändern.
Das trifft dann für alle Hardwaretypen zu weil man stets nur ein Device flashen bzw. bearbeiten kann.

- src_filter gehen, so ist es von dir ergänzt wurden, was ich unangetastet lies.

Die Ini Datei habe ich geprüft und auch alle einmal durchcompilieren lassen.

Reply:
Meine Frage bezog sich nur auf src_filter weil wir ihn nur bei dieserm einen Eintrag haben.

Offensichtlich brauchen wir den ja bei den anderen eintragen nicht. Vermutlich also auch nicht bei diesem.
Sieht wie ein Versuch aus, den ich anschließend nicht weiter verfolgt habe.